modules/dnn/src/layers/layer_norm.cpp

```diff
@@ -0,0 +1,160 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "layers_common.hpp"
+
+namespace cv { namespace dnn {
+
+class LayerNormLayerImpl CV_FINAL : public LayerNormLayer
+{
+public:
+    LayerNormLayerImpl(const LayerParams& params)
+    {
+        setParamsFrom(params);
+
+        // standard attr
+        axis = params.get<int>("axis", 0);
```

Comment:
Can you leave a comment why you chose 0?

Reply:
Actually, layer norm by definition can be performed on any axis. It always starts from the last axis to the chosen axis. For example, given input shape of [1, 50, 768], if axis=0, then layer norm is performed on all axes ([1, 50, 768]); if axis=1, then it is performed on [50, 768]; if axis=-1, then [768], etc...

For the vision transformers from torchvision, layer norm is performed on the last dim (axis=-1). However, given the fact that axis is normalized in importer, and we are not able to know the exact input dimensions in the constructor, I have to set it to 0 by default.

We can also normalize axis in `finalize` though if necessary.

Reply:
Oh, got it. I think it's fine with early axis normalization, just was wondering why it's not -1.